README.md

```diff
@@ -1,11 +1,123 @@
+
+### Step 8: Run and test sample iOS mobile application
+
+Run and test the iOS sample mobile application by following the steps in [Run and test sample iOS mobile application](https://learn.microsoft.com/entra/external-id/customers/how-to-run-native-authentication-sample-ios-app#run-and-test-sample-ios-mobile-application).
+
+### Step 9: Call a protected web API
+
+Follow the steps in Sign in users and call an API in a sample iOS mobile app by using native authentication to [sign in users and call a protected API in the iOS sample mobile app](https://learn.microsoft.com//entra/external-id/customers/sample-native-authentication-ios-sample-app-call-web-api).
+
+## Key concepts
+
+Open `NativeAuthSampleApp/Configuration.swift` file and you find the following lines of code:
+
+```swift
+import MSAL
+
+@objcMembers
+class Configuration: NSObject {
+    // Update the below to your client ID and tenantSubdomain you received in the portal.
+
+    static let clientId = "Enter_the_Application_Id_Here"
+    static let tenantSubdomain = "Enter_the_Tenant_Subdomain_Here"
+}
+```
+
+The code creates two constant properties:
+
+* _clientId_ - the value _Enter_the_Application_Id_Here_ is replaced with **Application (client) ID** of the app you register during the project setup. The **Application (client) ID** is unique identifier of your registered application.
+* _tenantSubdomain_ - the value _Enter_the_Tenant_Subdomain_Here_ is replaced with the Directory (tenant) subdomain. The tenant subdomain URL is used to construct the authentication endpoint for your app.
+
+You use `NativeAuthSampleApp/Configuration.swift` file to set configuration options when you initialize the client app in the Microsoft Authentication Library (MSAL).
+
+To create SDK instance, use the following code:
+
+```swift
+import MSAL
+
+var nativeAuth: MSALNativeAuthPublicClientApplication!
+
+do {
+    nativeAuth = try MSALNativeAuthPublicClientApplication(
+        clientId: Configuration.clientId,
+        tenantSubdomain: Configuration.tenantSubdomain,
+        challengeTypes: [.OOB, .password]
+    )
+} catch {
+    print("Unable to initialize MSAL \(error)")
+    showResultText("Unable to initialize MSAL")
+}
+```
+
+You create MSAL instance so that you can perform authentication logic and interact with your tenant through native authentication APIs. The `MSALNativeAuthPublicClientApplication` create an instance called `nativeAuth`. The configuration file `NativeAuthSampleApp/Configuration.swift` file is passed as parameter.For more information about SDK instance, see [Tutorial: Prepare your iOS app for native authentication](https://learn.microsoft.com/en-gb/entra/external-id/customers/tutorial-native-authentication-prepare-ios-app#create-sdk-instance)
```

Comment:
```suggestion
You create MSAL instance so that you can perform authentication logic and interact with your tenant through native authentication APIs. The `MSALNativeAuthPublicClientApplication` create an instance called `nativeAuth`. The `clientId` and `tenantSubdomain`, defined in theconfiguration file `NativeAuthSampleApp/Configuration.swift` file, are passed as parameter. For more information about SDK instance, see [Tutorial: Prepare your iOS app for native authentication](https://learn.microsoft.com/en-gb/entra/external-id/customers/tutorial-native-authentication-prepare-ios-app#create-sdk-instance)
```